socialdistribution/api/views.py

```diff
@@ -830,6 +831,69 @@ def author_profile(request, author_id):
     }
     return JsonResponse(response_body, status=405)
 
+@check_auth
+def github_posts(request, author_id):
+    if request.method == "GET":
+        author = Author.objects.get(id=author_id)
+
+        # get the github activities of the author
+        github_name = author.github.split("/")[3]
+        github_name = github_name.lower();
+        github_url = 'https://api.github.com/users/' + github_name + '/events'
+        r = requests.get(url = github_url)
+
+        if r.status_code != 200:
+            response_body = {
+                "query": "github_posts",
+                "success": False,
+                "message": "Can't retrieve the author's github activities",
+            }
+            return JsonResponse(response_body, status=r.status_code)
+
+        github_activities = r.json()
+        for activity in github_activities:
+
+            git_id = activity['id']
+            existing_post = Post.objects.filter(author=author, description=git_id).exists()
+            print(git_id)
+            if not existing_post:
+                title = activity['type']
+                published = activity['created_at']
+                type = activity['type']
+                repo_name = activity['repo']['name']
+                payload = activity['payload']
+
+                content = "<a class=\"post-card-text\" href=\"https://github.com/"+ repo_name + "\">"+ repo_name +"</a>"
```

Comment:
I'm not sure that the content should be raw html. I would suggest markdown, since you can add links and stuff but it's not vulnerable to XSS attacks.

Reply:
oh you are right, Changed them to markdown now